uvtools/dspec.py

```diff
@@ -1,78 +1,160 @@
+    Variable names preserved for backward compatability with capo/PAPER analysis.
+    
+    Arguments:
+        bl_len: length of baseline (in 1/[sdf], typically ns)
+        sdf: frequency channel width (typically in GHz)
+        nchan: number of frequency channels
+        standoff: fixed additional delay beyond the horizon (same units as bl_len)
+        horizon: proportionality constant for bl_len where 1 is the horizon (full light travel time)
+
+    Returns:
+        uthresh, lthresh: bin indices for upper and lower edges of the fourier filter
+    '''
     bl_dly = horizon * bl_len + standoff
-    bin_dly = 1. / (sdf * nchan)
-    w = int(round(bl_dly/bin_dly))
+    return calc_width(bl_dly, sdf, nchan)    
+
+
+def calc_width(filter_size, real_delta, nsamples):
+    '''Calculate the upper and lower bin indices of a fourier filter.
+
+    Arguments:
+        filter_size: the half-width (i.e. the width of the positive part) of the region in fourier 
+            space, symmetric about 0, that is filtered out. In units of 1/[real_delta].
+        real_delta: the bin width in real space
+        nsamples: the number of samples in the array to be filtered
+
+    Returns:
+        uthresh, lthresh: bin indices for upper and lower edges of the fourier filter    
+    '''
+    bin_width = 1. / (real_delta * nsamples)
+    w = int(round(filter_size / bin_width))
     uthresh, lthresh = w + 1, -w
-    if lthresh == 0: lthresh = nchan
-    return (uthresh,lthresh)
-    
-def delay_filter(data, wgts, bl_len, sdf, standoff=0., horizon=1., tol=1e-4, 
-        window='none', skip_wgt=0.5, maxiter=100):
-    '''Apply a wideband delay filter to data.  Data are weighted again by wgts,
-    windowed, Fourier transformed, and deconvolved allowing clean components
-    between lthresh and uthresh.  The mdl, residual, and info are returned in
-    frequency domain.'''
+    if lthresh == 0: 
+        lthresh = nsamples
+    return (uthresh, lthresh)
+
+
+def high_pass_fourier_filter(data, wgts, filter_size, real_delta, tol=1e-9, window='none', skip_wgt=0.1, maxiter=100):
+    '''Apply a highpass fourier filter to data. Uses aipy.deconv.clean. 
+
+    Arguments:
+        data: 1D or 2D (real or complex) numpy array to be filtered along the last dimension.
+        wgts: real numpy array of multiplicative weights with the same shape as the data. 
+        filter_size: the half-width (i.e. the width of the positive part) of the region in fourier 
+            space, symmetric about 0, that is filtered out. In units of 1/[real_delta].
+        real_delta: the bin width in real space of the dimension to be filtered
+        tol: CLEAN algorithm convergence tolerance (see aipy.deconv.clean)
+        window: window function for filtering applied to the filtered axis. 
+            See aipy.dsp.gen_window for options.
+        skip_wgt: skips filtering highly flagged rows (unflagged fraction ~< skip_wgt) 
```

Comment:
"Highly flagged" means the value of `wgts` is low, right?

Reply:
Clarified.